### ISSUE TITLE
ZipArchive::by_index errors

### ISSUE THREAD

**Issue by elpiel** [rzip/rzip#8]
The error I want to discuss in particular is the `Encrypted files are not supported`. Should we just skip them and collect all failed files, so we can extract the rest of them, or should we fail the whole extraction?

I am actually for the first option, i.e. collect all errors and let the process continue, we need to check also how `unzip` does it in such occasions.

**Comment by DavidBM** [comment]
Yeah. Me too. We should take care of the exit code in these cases. For example, in a CI you want to fail with error if you had warnings, but maybe in local you want to ignore warnings. 

Maybe we should look at unzip to see what they do.

**Comment by DavidBM** [comment]
#19 makes me think that the warnings are just printed and the exit code changes accordingly

**Comment by elpiel** [comment]
There are couple of candidates indeed, we should look into implementing some of the codes and also the mechanism for aborting (Ctrl+C) .
